gateway/api/views.py

```diff
@@ -890,3 +904,32 @@ def retrieve(self, request, pk=None):
                 instance, context={"user": user}
             )
         return Response(serializer.data)
+
+    @action(methods=["GET"], detail=False)
+    def by_title(self, request):
```

Comment:
Question: instead of having a separate endpoint to search by title, would it make sense to pass the search title as a parameter? something like `/api/v1/catalog?title=mytitle` ? Though it looks like we've already established a pattern with programs (`/programs` and `/programs/get_by_title`) so maybe its best to keep a consistent design...

Reply:
I think you have a point here @psschwei and I had the same doubt. The decision because I preferred this option was:
- `/api/v1/catalog?title=mytitle`: should return an `[]` of elements. So to maintain the coherence with the `retrieve` end-point and only return one function I preferred this approach.
- About `/programs/get_by_title` vs `by_title ` I think you are right and probably we should continue using `get_by_title` here too to maintain consistency at least in the `v1`.

Reply:
So basically this end-point should return only a `Function` not a `[Function]`. To have the same return as the `retrieve` by `id`.

Reply:
@akihikokuroda can we change `by_title ` by `get_by_title ` following @psschwei suggestion? 🙏